src/include/txn.i

```diff
@@ -881,7 +881,8 @@ __wt_txn_read_upd_list(
                  * Save the prepared update to help us detect if we race with prepared commit or
                  * rollback.
                  */
-                if (prepare_updp != NULL && *prepare_updp == NULL)
+                if (prepare_updp != NULL && *prepare_updp == NULL &&
+                  F_ISSET(upd, WT_UPDATE_PREPARE_RESTORED_FROM_DISK))
```

Comment:
This change is good. we need to restart only when the prepared update is restored from disk.

Reply:
Reverted and add more comments